database/sql/src/main/scala/cromwell/database/slick/WorkflowStoreSlickDatabase.scala

```diff
@@ -88,13 +88,25 @@ trait WorkflowStoreSlickDatabase extends WorkflowStoreSqlDatabase {
       } yield ()
     }
 
+    def fetchAndUpdateStartableWfs(hogGroup: Option[String]) = {
+      for {
+        workflowStoreEntries <- dataAccess.fetchStartableWfsForHogGroup(
+          (limit.toLong, heartbeatTimestampTimedOut, workflowStateExcluded, hogGroup)
+        ).result
+        _ <- DBIO.sequence(
+          workflowStoreEntries map updateForFetched(cromwellId, heartbeatTimestampTo, workflowStateFrom, workflowStateTo)
+        )
+      } yield workflowStoreEntries
+    }
+
     val action = for {
-      workflowStoreEntries <- dataAccess.fetchStartableWorkflows(
-        limit.toLong, heartbeatTimestampTimedOut, workflowStateExcluded, excludedGroups
-      ).result
-      _ <- DBIO.sequence(
-        workflowStoreEntries map updateForFetched(cromwellId, heartbeatTimestampTo, workflowStateFrom, workflowStateTo)
-      )
+      // find hog group with lowest count of actively running workflows
+      hogGroupOption <- dataAccess.getHogGroupWithLowestRunningWfs(heartbeatTimestampTimedOut, workflowStateExcluded, excludedGroups).result.headOption
```

Comment:
Random question: why are they called 'hog groups'?

Reply:
Haha I am not 100% sure, but I am assuming because these groups can "hog" Cromwell resources? 🤔 
Article explaining hog groups and hog factors: https://cromwell.readthedocs.io/en/develop/cromwell_features/HogFactors/